src/html/move-style-into-template.ts

```diff
@@ -45,24 +61,164 @@ class MoveStyleIntoTemplate extends HtmlRule {
           <dom-module>
   `);
 
-  async checkDocument(parsedDocument: ParsedHtmlDocument) {
-    const warnings: Warning[] = [];
-    const outOfPlaceStyle = p.AND(
-        p.hasTagName('style'), p.parentMatches(p.hasTagName('dom-module')));
-    const outOfPlaceStyles =
-        dom5.nodeWalkAll(parsedDocument.ast, outOfPlaceStyle);
-    for (const outOfPlaceNode of outOfPlaceStyles) {
-      warnings.push(new Warning({
-        parsedDocument,
-        code: this.code,
-        message:
-            `<style> tags should not be direct children of <dom-module>, they should be moved into the <template>`,
-        severity: Severity.WARNING,
-        sourceRange: parsedDocument.sourceRangeForNode(outOfPlaceNode)!
-      }));
+  async checkDocument(parsedDocument: ParsedHtmlDocument, document: Document) {
+    const warnings: FixableWarning[] = [];
+    const domModules = document.getFeatures({kind: 'dom-module'});
+    for (const domModule of domModules) {
+      const template = (domModule.astNode.childNodes ||
+                        []).find((n) => n.tagName === 'template');
+      if (!template) {
+        continue;
+      }
+      const moduleChildren = domModule.astNode.childNodes || [];
```

Comment:
nit: can be moved above at line 68, and be used to find the template

Reply:
+1, done